wallet/src/main/java/com/android/identity_credential/wallet/credman/CredmanPresentationActivity.kt

```diff
@@ -216,44 +142,53 @@ class CredmanPresentationActivity : FragmentActivity() {
                         Crypto.digest(Algorithm.SHA256, readerPublicKey.asUncompressedPointEncoding)
                     )
                 }
-                // Create ISO DeviceResponse
-                createMDocDeviceResponse(
-                    credentialId,
-                    dataElements,
-                    encodedSessionTranscript,
-                    onComplete = { deviceResponse, credential ->
-                        credential.increaseUsageCount()
-                        // The Preview protocol HPKE encrypts the response.
-                        val (cipherText, encapsulatedPublicKey) = Crypto.hpkeEncrypt(
-                            Algorithm.HPKE_BASE_P256_SHA256_AES128GCM,
-                            readerPublicKey,
-                            deviceResponse,
-                            encodedSessionTranscript
-                        )
-                        val encodedCredentialDocument =
-                            CredmanUtil.generateCredentialDocument(cipherText, encapsulatedPublicKey)
 
-                        // Create the preview response
-                        val responseJson = JSONObject()
-                        responseJson.put(
-                            "token",
-                            Base64.encodeToString(
-                                encodedCredentialDocument,
-                                Base64.NO_WRAP or Base64.URL_SAFE
-                            )
+                lifecycleScope.launch {
+                    val mdocCredential = getMdocCredentialForCredentialId(credentialId)
+                    // show Presentment FLow, skipping consent prompt
+                    val documentCborBytes = showPresentmentFlow(
+                        activity = this@CredmanPresentationActivity,
+                        walletApp = walletApp,
+                        documentRequest = DocumentRequest(dataElements),
+                        mdocCredential = mdocCredential,
+                        trustPoint = null,
```

Comment:
Please add a line above `trustpoint = null` that says: `TODO: Need to extend TrustManager with a verify() variants which takes a domain or appId`